src/handler/GenerateProjectHandler.ts

```diff
@@ -34,35 +41,32 @@ export class GenerateProjectHandler extends BaseHandler {
     public async runSteps(operationId?: string): Promise<void> {
 
         let step: IStep | undefined = SpecifyServiceUrlStep.getInstance();
-        const projectMetadata: IProjectMetadata = {
-            pickSteps: []
-        };
+
         SpecifyArtifactIdStep.getInstance().resetDefaultInput();
         SpecifyGroupIdStep.getInstance().resetDefaultInput();
         while (step !== undefined) {
-            step = await step.execute(operationId, projectMetadata);
+            step = await step.execute(operationId, this.metadata);
         }
 
-        this.metadata = projectMetadata;
-
         // Step: Choose target folder
-        this.outputUri = await instrumentOperationStep(operationId, "TargetFolder", specifyTargetFolder)(this.metadata.artifactId);
+        this.outputUri = await instrumentOperationStep(operationId, "TargetFolder", specifyTargetFolder)(this.metadata);
         if (this.outputUri === undefined) { throw new OperationCanceledError("Target folder not specified."); }
 
         // Step: Download & Unzip
         await instrumentOperationStep(operationId, "DownloadUnzip", downloadAndUnzip)(this.downloadUrl, this.outputUri.fsPath);
 
-        // Open in new window
-        const hasOpenFolder: boolean = (vscode.workspace.workspaceFolders !== undefined);
-        const candidates: string[] = [
-            "Open",
-            hasOpenFolder ? "Add to Workspace" : undefined,
-        ].filter(Boolean);
-        const choice: string = await vscode.window.showInformationMessage(`Successfully generated. Location: ${this.outputUri.fsPath}`, ...candidates);
-        if (choice === "Open") {
-            vscode.commands.executeCommand("vscode.openFolder", vscode.Uri.file(path.join(this.outputUri.fsPath, this.metadata.artifactId)), hasOpenFolder);
-        } else if (choice === "Add to Workspace") {
-            vscode.workspace.updateWorkspaceFolders(vscode.workspace.workspaceFolders.length, null, { uri: vscode.Uri.file(path.join(this.outputUri.fsPath, this.metadata.artifactId)) });
+        // Open project either is the same workspace or new workspace
+        const hasOpenFolder = vscode.workspace.workspaceFolders !== undefined || vscode.workspace.rootPath !== undefined;

```

Comment:
`vscode.workspace.rootPath` has been deprecated. 
see https://github.com/microsoft/vscode/blob/635cfbcd0f36bf3562b2f45a4995293869c15db2/src/vs/vscode.d.ts#L9754-L9760

Reply:
I have removed the reference to the deprecated `rootPath`.
The reason for it being there was that with **code-server** 3.4.1 with a folder opened by the code-server `rootPath` had a value and `workspaceFolders` was either empty or not set. Perhaps VSCode 1.45.1 still had the old root folder support in place...

Reply:
See the commit below